src/StreamingDataFrames/streamingdataframes/models/rows.py

```diff
@@ -23,14 +26,20 @@ class Row:
         "leader_epoch",
     )
 
+    # TODO: Maybe include headers here for if/when it's a dict?
+    _copy_map = {
+        "value": lambda self, k: deepcopy(getattr(self, k)),
+        "timestamp": lambda self, k: copy(getattr(self, k)),
+    }
+
     def __init__(
         self,
         value: Optional[dict],
-        topic: str,
-        partition: int,
-        offset: int,
-        size: int,
-        timestamp: MessageTimestamp,
+        topic: Optional[str] = None,
+        partition: Optional[int] = None,
+        offset: Optional[int] = None,
+        size: Optional[int] = None,
+        timestamp: Optional[MessageTimestamp] = None,
```

Comment:
This one is a bit more of a "suggestion" that I'm fine reverting: I thought this might help where you have to provide a bunch of these args that generally don't make sense, or arent really required, in the context of producing a new Row (my example app I made for show and tell comes to mind).

Might be a better way to do it, but this seemed like an easy thing to do in the interim. If you're not a fan we can revert it =)

Reply:
Reverting